lib/gui/etcher.js

```diff
@@ -20,31 +20,29 @@ const electron = require('electron')
 const path = require('path')
 const EXIT_CODES = require('../shared/exit-codes')
 const buildWindowMenu = require('./menu')
-let mainWindow = null
+const settings = require('./app/models/settings')
 
-electron.app.on('window-all-closed', electron.app.quit)
+/* eslint-disable lodash/prefer-lodash-method */
 
-// Sending a `SIGINT` (e.g: Ctrl-C) to an Electron app that registers
-// a `beforeunload` window event handler results in a disconnected white
-// browser window in GNU/Linux and macOS.
-// The `before-quit` Electron event is triggered in `SIGINT`, so we can
-// make use of it to ensure the browser window is completely destroyed.
-// See https://github.com/electron/electron/issues/5273
-electron.app.on('before-quit', () => {
-  process.exit(EXIT_CODES.SUCCESS)
-})
+const config = settings.getDefaults()
+let mainWindow = null
 
-electron.app.on('ready', () => {
+/**
+ * @summary Create Etcher's main window
+ * @example
+ * electron.app.on('ready', createMainWindow)
+ */
+const createMainWindow = () => {
   mainWindow = new electron.BrowserWindow({
     width: 800,
     height: 380,
     useContentSize: true,
     show: false,
-    resizable: Boolean(process.env.ETCHER_FULLSCREEN),
+    resizable: Boolean(config.fullscreen),
```

Comment:
Is this right? AFAICT the fact that you've got `const config = settings.getDefaults()` earlier on means that `config` will always be set to the values at https://github.com/resin-io/etcher/blob/d37ea80eff76fe2fdda767bdd4ed5c41f291e4e7/lib/gui/app/models/settings.js#L36 and thus `config.fullscreen` will _always_ be undefined, rather than the value that you've set in the actual config file?

Reply:
Oh, I've just seen the `settings.load()` part down at the bottom of the file :laughing: